src/crispy/Note.cpp

```diff
@@ -0,0 +1,68 @@
+#include <crispy/Note.h>
+#include <crispy/assert.h>
+
+#include <gsl/gsl_assert>
+
+#include <cmath>
+#include <vector>
+
+using namespace crispy;
+
+namespace
+{
+constexpr double SAMPLE_RATE = 44100;
+constexpr double square_wave(double x) noexcept
+{
+    x = std::fmod(x, 2);
+    return std::isgreater(x, 1) ? -1 : 1;
+}
+
+auto createNote(double volume, double duration, double frequency) noexcept
+{
+    duration = duration * 1 / 32.0;
+    volume /= 7;
+    std::vector<std::int16_t> buffer;
+    buffer.reserve(static_cast<size_t>(std::ceil(duration * SAMPLE_RATE)));
+    for (double i = 0; i < duration * SAMPLE_RATE; ++i)
+    {
+        auto sample = static_cast<int16_t>(0x7fff * volume * square_wave(frequency / SAMPLE_RATE * i * 2));
+        buffer.push_back(sample);
+    }
+    return buffer;
+}
+
+} // namespace
+std::vector<std::int16_t> crispy::createNote(double volume, int duration, Note note_) noexcept
+{
+    Ensures(static_cast<int>(note_) >= 0 && static_cast<int>(note_) < 26);
+    switch (note_)
+    {
+        case Note::A5: return ::createNote(volume, duration, 880.0); break;
+        case Note::A6: return ::createNote(volume, duration, 1760.0); break;
+        case Note::A_SHARP_5: return ::createNote(volume, duration, 932.3275); break;
```

Comment:
This switch can be greatly simplified as the formula is is something like `2 ^ something` (I forgot), but it's a power of two with the base node concert A (A4) being 442 Hz, nowadays. I'm just dropping of my brain here, it's okay to have it like that for now and do that kind of algorithmic simplification later. :)